diff --git a/browser/components/sessionstore/TabGroupState.sys.mjs b/browser/components/sessionstore/TabGroupState.sys.mjs
index 934326aefc80479a2ca2ee9e376375f117512b4d..2ed703acc8fe38edd521a30f65de56218531a843 100644
--- a/browser/components/sessionstore/TabGroupState.sys.mjs
+++ b/browser/components/sessionstore/TabGroupState.sys.mjs
@@ -62,6 +62,11 @@ class _TabGroupState {
    */
   collect(tabGroup) {
     return {
+      pinned: tabGroup.pinned,
+      essential: tabGroup.essential,
+      splitView: tabGroup.hasAttribute("split-view-group"),
+      parentGroupId: tabGroup.group?.id,
+      isZenFolder: tabGroup.isZenFolder,
       id: tabGroup.id,
       name: tabGroup.label,
       color: tabGroup.color,
